PWGHF/D2H/TableProducer/dataCreatorCharmHadPiReduced.cxx

```diff
@@ -936,9 +963,44 @@ struct HfDataCreatorCharmHadPiReduced {
             }
           }
         }
-        rowHfLcPiMcCheckReduced(pdgCodeBeautyMother, pdgCodeCharmMother, pdgCodeProng0, pdgCodeProng1, pdgCodeProng2, pdgCodeProng3);
+        tables.rowHfLcPiMcCheckReduced(pdgCodeBeautyMother, pdgCodeCharmMother, pdgCodeProng0, pdgCodeProng1, pdgCodeProng2, pdgCodeProng3);
+      }
+      tables.rowHfLcPiMcRecReduced(indexHfCandCharm, selectedTracksPion[vecDaughtersB.back().globalIndex()], flag, flagWrongCollision, debug, motherPt);
+    } else if constexpr (decChannel == DecayChannel::B0ToDstarPi) {
+      // B0 → D*+ π- → (D0 π+) π- → (K- π+ π+) π-
+      auto indexRec = RecoDecay::getMatchedMCRec<true, false, false, true, true>(particlesMc, std::array{vecDaughtersB[0], vecDaughtersB[1], vecDaughtersB[2], vecDaughtersB[3]}, Pdg::kB0, std::array{-kKPlus, +kPiPlus, +kPiPlus, -kPiPlus}, true, &sign, 4);
+      if (indexRec > -1) {
+        // D*+ → (D0 π+) → K- π+ π+
```

Comment:
In case of the `D*` it would also be good to match the D0 daughter (it should never happen that, if the `D*` is matched the `D0` is not, but just to be sure)